### PR TITLE
Use an existing bazel-builder image instead of building it every time

### PR DESCRIPTION
**Pull request by aglitke**
Signed-off-by: Adam Litke <alitke@redhat.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by rmohr** [comment]
> So this is going to be a chicken and egg problem, we need to push the builder to dockerhub first, and then we can use it, but not before. I see you 'solved' it by using your own. kubevirt has a separate 'push' for just that one image outside of bazel. Something to look into?

This is indeed tricky. We have a pretty complex solution right now. We automatically push the base image after every merge for every branch. So if something is merged against `master`, we push to the `:master` tag.

The tests in prow just pull dumb the latest one of the target branch of a PR.

I see this system as a technical debt. Most of the things which we install in the build container are still coming from the pre-bazel time, or are easy to replace. At the end we should pretty much only need bazel and rsync in there.

File: hack/build/config.sh [hunk 1]
```diff
@@ -24,6 +24,7 @@ FUNC_TEST_HTTP="cdi-func-test-file-host-http"
 FUNC_TEST_REGISTRY="cdi-func-test-registry"
 FUNC_TEST_REGISTRY_POPULATE="cdi-func-test-registry-populate"
 FUNC_TEST_REGISTRY_INIT="cdi-func-test-registry-init"
+BUILDER_TAG=${BUILDER_TAG:="kubevirt/kubevirt-cdi-bazel-builder"}
```
Comment (awels) [inline hack/build/config.sh]: Can you make this:
```
BUILDER_TAG=${BUILDER_TAG:-kubevirt-cdi-bazel-builder@sha256:38e449bb8f5e2e3dc8ae241323c69ce37e72bd98a109d2032955dd9bd0e0a545}
```

File: hack/build/config.sh [hunk 2]
```diff
@@ -24,6 +24,7 @@ FUNC_TEST_HTTP="cdi-func-test-file-host-http"
 FUNC_TEST_REGISTRY="cdi-func-test-registry"
 FUNC_TEST_REGISTRY_POPULATE="cdi-func-test-registry-populate"
 FUNC_TEST_REGISTRY_INIT="cdi-func-test-registry-init"
+BUILDER_TAG=${BUILDER_TAG:-kubevirt-cdi-bazel-builder@sha256:38e449bb8f5e2e3dc8ae241323c69ce37e72bd98a109d2032955dd9bd0e0a545}
```
Comment (awels) [inline hack/build/config.sh]: Okay I messed up when I gave the info, it should be:
```
BUILDER_TAG=${BUILDER_TAG:-kubevirt/kubevirt-cdi-bazel-builder@sha256:38e449bb8f5e2e3dc8ae241323c69ce37e72bd98a109d2032955dd9bd0e0a545}
```
Note the extra 'kubevirt'

File: hack/build/in-docker.sh [hunk 3]
```diff
@@ -18,10 +18,10 @@ set -e
 
 script_dir="$(readlink -f $(dirname $0))"
 source "${script_dir}"/common.sh
+source "${script_dir}"/config.sh
 
 WORK_DIR="/go/src/kubevirt.io/containerized-data-importer"
 BUILDER_SPEC="${BUILD_DIR}/docker/builder"
-BUILDER_TAG="kubevirt-cdi-bazel-builder"
 
 # Build the encapsulated compile and test container
```
Comment (awels) [inline hack/build/in-docker.sh]: These two lines need to be removed as well, its still trying to build these.

**Pull request merged**